### PR TITLE
Fix code blocks indents in .md files

### PR DESCRIPTION
**Pull request by fanzhuyifan**
Previously a lot of the code blocks in markdown files were horribly indented with bad white spaces in the beginning of lines. Users can't just select, copy, paste, and run (in the case of python). I have fixed all these, so there is no longer any code block with bad white space at the beginning of the lines.
It would be nice if you could try to make sure in future commits that **all code blocks are properly indented inside and have the right amount of white space in the beginning**!

### CODE REVIEW

**Comment by fanzhuyifan** [comment]
This is a follow up to [this pull request](https://github.com/intel-analytics/analytics-zoo/pull/2976).

File: docs/docs/APIGuide/FeatureEngineering/image.md [hunk 1]
```diff
@@ -50,9 +50,9 @@ def rdd(data: RDD[ImageFeature]): DistributedImageSet
 Create DistributedImageSet from rdd of ImageFeature
 
 * data: array of ImageFeature
-```
+  ```
```
Comment (qiyuangong) [inline docs/docs/APIGuide/FeatureEngineering/image.md:53]: Well, I guess this is a mis-modify. Pls keep them in the same style.

**Comment by qiyuangong** [comment]
Nice PR! This PR is very helpful for our document. Thank you @fanzhuyifan 👍 

One small suggestion about this PR: if PR changes too many files (more than 10 files), pls split it into several ones.

Add @hkvision for style review.

File: docs/docs/APIGuide/FeatureEngineering/image.md (hunk 1 above)
Comment (fanzhuyifan) [reply docs/docs/APIGuide/FeatureEngineering/image.md]: Yeah you are right. Fixed

**Comment by fanzhuyifan** [comment]
Ok sure! I will remember to do that in the future!
Do you mean split into several commits or split into several PRs?

**Comment by qiyuangong** [comment]
> Ok sure! I will remember to do that in the future!
> Do you mean split into several commits or split into several PRs?

Several PR. Small PR is easy to Review/Merge, while large PR may leads to long review period or discussion.

File: docs/docs/APIGuide/PipelineAPI/nnframes.md [hunk 2]
```diff
@@ -437,22 +437,22 @@ For the complete examples of NNFrames, please refer to:
 into DataFrame.
 
 Scala:
-```scala
+   ```scala
     val imageDF = NNImageReader.readImages(imageDirectory, sc)
```
Comment (hkvision) [inline docs/docs/APIGuide/PipelineAPI/nnframes.md:441]: I think ```scala would be unchanged; instead remove the tab at the beginning of the code?

File: docs/docs/APIGuide/PipelineAPI/nnframes.md [hunk 3]
```diff
@@ -437,22 +437,22 @@ For the complete examples of NNFrames, please refer to:
...
+   ```
 
 The output DataFrame contains a sinlge column named "image". The schema of "image" column can be
 accessed from `com.intel.analytics.zoo.pipeline.nnframes.DLImageSchema.byteSchema`.
 Each record in "image" column represents one image record, in the format of
 Row(origin, height, width, num of channels, mode, data), where origin contains the URI for the image file,
 and `data` holds the original file bytes for the image file. `mode` represents the OpenCV-compatible
 type: CV_8UC3, CV_8UC1 in most cases.
-```scala
+  ```scala
```
Comment (hkvision) [inline docs/docs/APIGuide/PipelineAPI/nnframes.md:455]: same as above for this file

File: zoo/src/main/scala/com/intel/analytics/zoo/examples/recommendation/README.md [hunk 4]
```diff
@@ -19,7 +19,7 @@ In NeuralCFexample and WideAndDeepExample, we demostrate how to use Analyticd Zo
    WND Learning Model, proposed by ([Google, 2016](https://arxiv.org/pdf/1606.07792.pdf)), is a DNN-Linear mixed model. WND combines the strength of memorization and generalization. It's useful for generic large-scale regression and classification problems with sparse input features(e.g., categorical features with a large number of possible feature values). It has been used for Google App Store for their app recommendation.
 ### Run the wide and deep example with ml-1m dataset
 ``` bash
-   export ANALYTICS_ZOO_HOME=the folder where you extract the downloaded Analytics Zoo zip package
+export ANALYTICS_ZOO_HOME=the folder where you extract the downloaded Analytics Zoo zip package
    master=... // spark master
    ${ANALYTICS_ZOO_HOME}/bin/spark-shell-with-zoo.sh \
```
Comment (hkvision) [inline zoo/src/main/scala/com/intel/analytics/zoo/examples/recommendation/README.md:24]: These two lines should be modified as well :)

File: zoo/src/main/scala/com/intel/analytics/zoo/examples/recommendation/README.md [hunk 5]
```diff
@@ -32,7 +32,7 @@ In NeuralCFexample and WideAndDeepExample, we demostrate how to use Analyticd Zo
 
 ### Run the wide and deep example with Census Income dataset
 ``` bash
-   export ANALYTICS_ZOO_HOME=the folder where you extract the downloaded Analytics Zoo zip package
+export ANALYTICS_ZOO_HOME=the folder where you extract the downloaded Analytics Zoo zip package
    master=... // spark master
```
Comment (hkvision) [inline zoo/src/main/scala/com/intel/analytics/zoo/examples/recommendation/README.md:36]: same as above

**Review by qiyuangong — APPROVED** [review]
LGTM

**Comment by hkvision** [comment]
Thanks @fanzhuyifan so much! I will do the review for all files and you don't need to split into separate files this time 🥇

File: docs/docs/APIGuide/PipelineAPI/nnframes.md (hunk 2 above)
Comment (fanzhuyifan) [reply docs/docs/APIGuide/PipelineAPI/nnframes.md]: Fixed

File: docs/docs/APIGuide/PipelineAPI/nnframes.md (hunk 3 above)
Comment (fanzhuyifan) [reply docs/docs/APIGuide/PipelineAPI/nnframes.md]: Fixed

File: zoo/src/main/scala/com/intel/analytics/zoo/examples/recommendation/README.md (hunk 4 above)
Comment (fanzhuyifan) [reply zoo/src/main/scala/com/intel/analytics/zoo/examples/recommendation/README.md]: Could you take a look if this [commit](https://github.com/intel-analytics/analytics-zoo/pull/2978/commits/2b5323cc0d7026b3b420054ede9d4c7f3c8ebd01) is good?

Change indent from 3 spaces to 4, and add "\" for multiline bash commands

File: zoo/src/main/scala/com/intel/analytics/zoo/examples/recommendation/README.md (hunk 5 above)
Comment (fanzhuyifan) [reply zoo/src/main/scala/com/intel/analytics/zoo/examples/recommendation/README.md]: See above

**Comment by fanzhuyifan** [comment]
I just realized I have not checked that these changes work under the html renderer used to generate the website. They work for the build-in Github viewer, but may not work for the website generator. See [here](https://analytics-zoo.github.io/master/#ProgrammingGuide/pytorch/) for an example of a break

File: docs/docs/KerasStyleAPIGuide/Layers/pooling.md [hunk 6]
```diff
@@ -787,35 +787,35 @@ input = np.random.random([2, 2, 2, 2, 3])
 output = model.forward(input)
 ```
 Input is:
-```python
-      [[[[[ 0.8402289  0.11503692 0.27831015]
-          [ 0.45756199 0.15043262 0.78778086]]
+  ```python
```
Comment (hkvision) [inline docs/docs/KerasStyleAPIGuide/Layers/pooling.md:790]: revert this line back and just remove the spaces before [[?

File: docs/docs/ProgrammingGuide/workingwithimages.md [hunk 7]
```diff
@@ -36,7 +36,7 @@ Each record in "image" column represents one image record, in the format of
 Row(origin, height, width, num of channels, mode, data), where origin contains the URI for the image file,
 and `data` holds the original file bytes for the image file. `mode` represents the OpenCV-compatible
 type: CV_8UC3, CV_8UC1 in most cases.
-```scala
+  ```scala
```
Comment (hkvision) [inline docs/docs/ProgrammingGuide/workingwithimages.md:39]: same as above

File: docs/docs/ProgrammingGuide/workingwithimages.md [hunk 8]
```diff
@@ -46,7 +46,7 @@ type: CV_8UC3, CV_8UC1 in most cases.
       StructField("mode", IntegerType, false) ::
       // Bytes in OpenCV-compatible order: row-wise BGR in most cases
       StructField("data", BinaryType, false) :: Nil)
-```
+  ```
```
Comment (hkvision) [inline docs/docs/ProgrammingGuide/workingwithimages.md:49]: ??

File: docs/docs/ScalaUserGuide/run.md [hunk 9]
```diff
@@ -51,7 +51,7 @@ You can run a analytics zoo program, e.g., the [Object Detection](https://github
 1. Download the pre-trained model from [here](https://github.com/intel-analytics/analytics-zoo/blob/master/docs/docs/ProgrammingGuide/object-detection.md).
 2. Prepare predict images
 3. Run the following command:
-```bash
+  ```bash
```
Comment (hkvision) [inline docs/docs/ScalaUserGuide/run.md:54]: same as above

File: zoo/src/main/scala/com/intel/analytics/zoo/examples/recommendation/README.md [hunk 10]
```diff
@@ -19,65 +19,65 @@ In NeuralCFexample and WideAndDeepExample, we demostrate how to use Analyticd Zo
...
-   master=... // spark master
-   ${ANALYTICS_ZOO_HOME}/bin/spark-shell-with-zoo.sh \
-   --master $master \
-   --driver-memory 4g \
-   --executor-memory 4g \
-   --class com.intel.analytics.zoo.examples.recommendation.WideAndDeepExample \
-   --inputDir ./data/ml-1m \
-   --dataset ml-1m
+export ANALYTICS_ZOO_HOME=the folder where you extract the downloaded Analytics Zoo zip package \
+    master=spark master \
```
Comment (hkvision) [inline zoo/src/main/scala/com/intel/analytics/zoo/examples/recommendation/README.md:23]: I don't think `\` is needed for these two lines?
Users can't directly paste these two lines but need to set properly according to their environment.
These two lines should be executed separately first followed by spark-shell

File: zoo/src/main/scala/com/intel/analytics/zoo/examples/recommendation/README.md [hunk 11]
```diff
@@ -19,65 +19,65 @@ In NeuralCFexample and WideAndDeepExample, we demostrate how to use Analyticd Zo
...
+    ${ANALYTICS_ZOO_HOME}/bin/spark-shell-with-zoo.sh \
```
Comment (hkvision) [inline zoo/src/main/scala/com/intel/analytics/zoo/examples/recommendation/README.md:24]: remove the space this line?

File: zoo/src/main/scala/com/intel/analytics/zoo/examples/recommendation/README.md [hunk 12]
```diff
@@ -19,65 +19,65 @@ In NeuralCFexample and WideAndDeepExample, we demostrate how to use Analyticd Zo
...
-   --master $master \
-   --driver-memory 4g \
-   --executor-memory 4g \
-   --class com.intel.analytics.zoo.examples.recommendation.WideAndDeepExample \
-   --inputDir ./data/census \
-   --batchSize 320 \
-   --maxEpoch 20 \
-   --dataset census
+export ANALYTICS_ZOO_HOME=the folder where you extract the downloaded Analytics Zoo zip package \
+    master=spark master \
```
Comment (hkvision) [inline zoo/src/main/scala/com/intel/analytics/zoo/examples/recommendation/README.md:36]: same as above

**Comment by hkvision** [comment]
jenkins passed: http://10.239.47.210:18888/job/ZOO-PR-Validation/4524/

**Comment by hkvision** [comment]
Thanks so much for your contributions @fanzhuyifan 
For the remaining several minor issues in this PR, we would take over to modify them :)

**Comment by fanzhuyifan** [comment]
> 
> 
> Thanks so much for your contributions @fanzhuyifan
> For the remaining several minor issues in this PR, we would take over to modify them :)

Thanks! Here's what happened:
1. I realized my changes might break the website because the md renderer you are using is different from the default md renderer GitHub uses or the md renderer am I using.
1. To make sure I am not breaking things, I wanted to test by generating the html files the way you are generating them. So I tried to generate the html pages using the instructions [here](https://github.com/intel-analytics/analytics-zoo/tree/master/docs). But
    1. By default it pulls the origin/master branch and uses that for html generation, which is unfortunate, since obviously I want to test my own local branch.
    1. It uses a directory `/tmp/zoo-doc/` hard-coded into the python file, which is also unfortunate.
1. After I sorted those things out and got `gen_site.py` happily running on my local version of files, it asked me to install spark! But I don't want to install spark just to render some markdown files into html pages.
1. So I gave up.

**Comment by hkvision** [comment]
Hi @fanzhuyifan I totally understand your points. Seems we are using Spark for rendering Python docs lol. Never mind and we have followed your contributions to further polish our docs. Thank so much and look forward to your future commits!

**Pull request merged**